### DISCUSSION TITLE
the same sql return different result when i connect to mysql database or h2 database

### DISCUSSION THREAD

**Discussion by baizhi1989** [dbeaver/dbeaver#15140] [General]
#### System information: 
- Operating system (distribution) and version win10
- DBeaver version 21.3.0.202111281534
- Additional extensions

#### Connection specification:
- Database name and version mysql  mysql-8.0.23
- Driver name mysql 5
- Do you use tunnels or proxies (SSH, SOCKS, etc)?  no
- Database name and version h2 database h2-1.4.199
- Driver name h2-1.4.199
- Do you use tunnels or proxies (SSH, SOCKS, etc)?  no

#### Describe the problem you're observing:
 the same sql  return different result when i connect to mysql database or h2 database
#### Steps to reproduce, if exist:
CREATE TABLE `a` (
  `c1` varchar(100) DEFAULT NULL,
  `c2` varchar(100) DEFAULT NULL,
  `c3` varchar(100) DEFAULT NULL
) ;

INSERT INTO a (c1,c2,c3) VALUES  ('1','a\\b','c');

-- return nothing when connect to mySql database,  press \ onece
select * from a where c2 like '%a\b%';
-- return nothing when connect to mySql database,  press \ twice
select * from a where c2 like '%a\\\\b%';

-- return nothing when connect to H2 database, press \ onece
select * from  a where c2 like '%a\b%';
-- **return record** in when connect to database,  press \ twice
select * from  a where c2 like '%a\\\\b%';

**Comment by ShadelessFox** [comment]
Hello @baizhi1989,

As you mentioned, it depends on the database you're connected to.

We do not process SQL in any way, but the database may process your query differently.

**Comment by LonwoLonwo** [comment]
Also, MySQL Workbench and H2 console return the same result as you mentioned.

![2022-01-10 12_29_34-MySQL Workbench](https://user-images.githubusercontent.com/45152336/148749862-4c7abce0-19ef-41e1-b399-7f2f868cbc2b.png)

![2022-01-10 12_30_00-H2 Console — Mozilla Firefox](https://user-images.githubusercontent.com/45152336/148749866-79b43281-9088-4750-b21b-65c4abba4917.png)

![2022-01-10 12_30_21-H2 Console — Mozilla Firefox](https://user-images.githubusercontent.com/45152336/148749867-f195cc52-7b3d-42f6-95fb-7e29636397ee.png)
